ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/sharing/AppSharingPresenter.java

```diff
@@ -82,24 +89,21 @@ public void onFailure(Throwable caught) {
 
         @Override
         public void onSuccess(String result) {
-            JSONArray permissionsArray =
-                    jsonUtil.getArray(jsonUtil.getObject(result), "apps"); //$NON-NLS-1$
-            if (permissionsArray != null) {
-                sharingList = new FastMap<>();
-                for (int i = 0; i < permissionsArray.size(); i++) {
-                    JSONObject user_perm_obj = permissionsArray.get(i).isObject();
-                    String id = jsonUtil.getString(user_perm_obj, "id"); //$NON-NLS-1$
-                    String appName = jsonUtil.getString(user_perm_obj,"name");
-                    JSONArray user_arr = jsonUtil.getArray(user_perm_obj, "permissions"); //$NON-NLS-1$
-
-                    loadPermissions(id,appName, user_arr);
-                }
-
-                final List<String> usernames = new ArrayList<>();
-                usernames.addAll(sharingList.keySet());
-                collaboratorsUtil.getUserInfo(usernames, new GetUserInfoCallback(usernames));
+            AutoBean<AppUserPermissionsList> abrp =
+                    AutoBeanCodex.decode(shareFactory, AppUserPermissionsList.class, result);
+            AppUserPermissionsList appPermsList = abrp.as();
+            sharingList = new FastMap<>();
```

Comment:
The next refactor that could be done is moving some of these decoding steps to a callback converter so the service returns the `AppUserPermissionsList`, for example, directly instead of a String.  Not sure if refactoring the appropriate methods in `appService` to use a callback converter might result in refactors needed elsewhere though.